virtualmachine/vsphere/util.go

```diff
@@ -419,13 +419,23 @@ var reconfigureVM = func(vm *VM, vmMo *mo.VirtualMachine) error {
 		return err
 	}
 
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
+	}
+
+	dsMo, err := findDatastore(vm, dcMo, vm.Datastores[0])
```

Comment:
Missing error handling for vm.Datastores[0].
Is it assumed to be handle in caller? 
Generic: For all param validations, if it is assumed to be handle by caller, Add a comment to inform that explicitly.